### ISSUE TITLE
False Positive with subclassed Enums

### ISSUE THREAD

**Issue by jonapich** [PyCQA/pylint#2626]
Adding custom methods to an Enum subclass and then implementing it: pylint can't see the new methods.

```
from enum import Enum, auto


class TestBase(Enum):
    """ Adds a special method to enums. """

    def hello_pylint(self) -> str:
        """ False positive. """
        return self.name


class TestEnum(TestBase):
    """ Tests the false positive for enums. """
    a = auto()
    b = auto()


test_enum = TestEnum.a
assert test_enum.hello_pylint() == test_enum.name
```

> [no-member] Instance of 'auto' has no 'hello_pylint' member
> [no-member] Instance of 'auto' has no 'name' member

Similar to #1932 and #2062

**Comment by PCManticore** [comment]
Thanks for the report!

**Comment by belm0** [comment]
I believe this problem happens regardless of subclassing.

What will it take for pylint/astroid to be aware of Enum's patching of class members?

**Comment by bhargavrpatel** [comment]
Just adding more examples on this as it may help close this issue.

```python
import typing
from enum import Enum


class MyEnum(Enum):
    def to_pb2(self) -> int:
        return typing.cast(int, self.value)

    def __str__(self) -> str:
        return self.name.upper()  # pylint: disable=no-member


class SomeEnum(MyEnum):
    A = "A"


class AnotherEnum(Enum):
    B = "B"


SomeEnum.__members__.items()
AnotherEnum.__members__.items()
```

This will report an issue for SomeEnum.__members__

**Comment by belm0** [comment]
@nelfin do you have a PR forthcoming?  https://github.com/nelfin/astroid/commit/afd0c2bbd56658228a24c453333f38eef6f5052c

**Comment by Pierre-Sassoulas** [comment]
Will be available in pylint [2.9.0 (see milestone for release planning)](https://github.com/PyCQA/pylint/milestone/25), or you can have it with pylint 2.8.2 if you set the astroid version to 2.5.8 yourself.

**Comment by belm0** [comment]
@nelfin I'm actually still seeing this problem with subclassed enums on the latest astroid (2.5.8) and pylint pre-release.  Has anyone confirmed otherwise?

**Comment by nelfin** [comment]
Ah, now I understand. The commits we've been talking about actually only address the issue in the comment by bhargavrpatel above (https://github.com/PyCQA/pylint/issues/2626#issuecomment-627442384), which talks about `Enum.__members__`. I'd misread the issue when I was first looking for duplicates about false-positives on `Enum.__members__`. I don't know if any work has been done to address incorrect inference of attributes on `enum.Enum` subclasses.

**Comment by belm0** [comment]
If there's any guidance you can offer on what this problem might be or what parts of the implementation to poke at, I can try looking at it.

**Comment by nelfin** [comment]
@belm0, I think [`astroid/brain/brain_namedtuple_enum.py:infer_enum_class`](https://github.com/PyCQA/astroid/blob/1342591e2beb955a377e4486e5595478f79789e8/astroid/brain/brain_namedtuple_enum.py#L184) would probably be the most likely place to start
